fixed bug of handlers and updated the testcases.

abhayymishraa opened the pull request:
<!-- Thanks for contributing to OWASP Nest!-->

<!-- Don't forget to link your PR to an existing issue if any.-->
Resolves #659 

<!-- Describe the big picture of your changes.-->
Add the PR description here.
- [x] Fixed bug 
and improved coverage 
<!-- Thanks again for your contribution!-->
 i teseted and made sure testcases work
![Screenshot from 2025-01-29 23-40-38](https://github.com/user-attachments/assets/f40280c6-06bc-40ca-8cd4-4c1dfb9487a4)

arkid15r reviewed (commented):
Good catch!

I'll add an extra point if you implement a solution that will make sure we avoid this type of problems in future (hint: pre-commit)

abhayymishraa commented:
> Good catch!
> 
> I'll add an extra point if you implement a solution that will make sure we avoid this type of problems in future (hint: pre-commit)

i was thinking about the same either we can make pre-commit or a workflow , we can use both !!!

abhayymishraa commented:
Hey @arkid15r 
In this latest commit i have implemeted the following chnages :
- Added a new pre-commit hook that makes sure that testcases file should be ending with ```*_test.py```
(there is a reason for this chnages as the official pre-commit hooks already have a hook to check it  but we are using  files in this format ```*_tests``` which is not the standard and in order to implement it without this we had to make a python file ourselve to check which it not optimal that's why i choose this approach)

- Replaced  ```*_tests.py```  to ```*_test.py```
- Removed constant.py inside test as it was failing in pre-commit and also it is of no use!!(i think so)

After this commit :
I have made sure it will pass for the right file formats:

  -  ![Screenshot from 2025-01-30 15-23-27](https://github.com/user-attachments/assets/614b2122-fbee-47d0-ba3c-af67be67c8b2)
  
i have made sure it fails when it find testcase externsion incorrect :
  
- ![Screenshot from 2025-01-30 15-23-59](https://github.com/user-attachments/assets/0bf962ec-b31e-4890-8f17-f7ad4ca2c27b)

i have made sure all testcase are working with same coverage 
- ![Screenshot from 2025-01-30 15-09-26](https://github.com/user-attachments/assets/3b6b9d00-01e4-491c-b62e-a5cb5f5fc8de)

abhayymishraa commented:
Your Welcome!!!!
